src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -291,14 +241,10 @@ private void DrainQueue()
             // Tell the queue not to accept any more items
             _queue.Complete();
 
-            // Spin through the queue and pass in our cancelled token, so that the waiting tasks are cancelled.
-            // NOTE: This only really works because the first thing that CallbackAsync does is check for cancellation
-            // but generics make it annoying to store the TaskCompletionSource<TResult> on the QueueItem so this
-            // is the best we can do for now. Ideally we would manipulate the TaskCompletionSource directly here
-            // and just call SetCanceled
+            // Spin through the queue and cancel all waiting tasks.
             while (_queue.TryDequeue(out var item))
             {
-                _ = item.CallbackAsync(null, new CancellationToken(true));
+                _ = item.TrySetCanceled(_cancelSource.Token);
```

Comment:
something feels weird here with the CTs.   it may be right.  but i'm not feeling confident it it.  I feel like we should not be cancelling it with our token, but shuld just be asking the item to cancel itself with it's own token that tracks its work.

--

note: i'd liek to actually talk through this so I can get a concreate understanding of exactly how cancellation works here.  thanks :)